libs/frontend/util-tooltip/src/lib/use-tooltip.ts

```diff
@@ -45,51 +55,94 @@ export const useTooltip = (options?: IUseTooltipOptions): IUseTooltip => {
     middleware.push(arrow({ element: arrowRef }))
   }
 
-  const itemRef = ref(null)
+  const itemRef = ref<Element | null>(null)
   const tooltipRef = ref(null)
   const show = ref(false)
   const tooltip = useFloating(itemRef, tooltipRef, {
     placement,
     middleware,
   })
-  const side = placement.split('-')[0]
-  const { x, y, strategy, update, middlewareData } = tooltip
+  const { x, y, strategy, update } = tooltip
 
   if (options?.arrow) {
-    const arrowX = computed(() => middlewareData.value.arrow?.x ?? null)
-    const arrowY = computed(() => middlewareData.value.arrow?.y ?? null)
+    arrowStyle.value = {
+      left: '0px',
+      right: '0px',
+    }
+  }
+
+  interface IArrow {
+    x?: number
+    y?: number
+  }
+
+  const computeArrow = (arrow: IArrow, placement: string) => {
+    if (arrowStyle.value) {
+      const side = placement.split('-')[0]
 
-    arrowStyle = computed(() => {
+      const x = arrow.x ?? null
+      const y = arrow.y ?? null
       const staticSide =
         {
           top: 'bottom',
           right: 'left',
           bottom: 'top',
           left: 'right',
         }[side] ?? placement
-      const offset = options?.offset ?? 2
-      return {
-        left: arrowX.value != null ? `${arrowX.value}px` : '',
-        top: arrowY.value != null ? `${arrowY.value}px` : '',
+      const offset = options?.offset ?? 10
+      arrowStyle.value = {
+        left: x === null ? '' : `${x}px`,
+        top: y === null ? '' : `${y}px`,
         [staticSide]: `-${offset - 2}px`,
       }
-    })
+    }
   }
 
-  const tooltipStyle = computed(() => ({
+  const tooltipStyle = ref<ITooltipStyle>({
     position: strategy.value,
     top: `${y.value ?? 0}px`,
     left: `${x.value ?? 0}px`,
     width: 'max-content',
     'z-index': 1001,
-  }))
+  })
+
+  const updatePosition = async () => {
+    if (itemRef.value && tooltipRef.value) {
```

Comment:
@abemscac I rewrote `use-tooltip` to make use of `autoUpdate`, because I couldn't figure out why all the tooltips were suddenly placed incorrectly. For some reason, `useFloating` resulted in top/left = 0, where before it worked correctly.

This more closely matches floating-ui examples for arrow styling, so I think it's ok, even though I don't understand how the previous code broke.